src/components/Extension/CalDotComExtension/BookAppointmentAction.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { FC, useCallback, useMemo } from 'react'
+import { useTranslation } from 'next-i18next'
+import { CalDotComScheduling } from '@awell_health/ui-library'
+import { mapActionFieldsToObject } from '../utils'
+
+import { BookAppointmentFields } from './types'
+import type { ExtensionActivityRecord } from '../../../types/generated/types-orchestration'
```

Comment:
Avoid using types from the generated code directly. The recommended alternative is to:
- Export types from the `./types.ts` file.
- Use types from the actual query document instead of the generic type so that it properly reflects the data being queried

So in this case you would export the extension activity record like this: (not tested so you might need to tweak this a bit)
```
export type ExtensionActivityRecord = GetExtensionActivityDetailsQuery['pluginActivityRecord']['record']
```

Reply:
Good tip, thanks!